File: day9.js
import { readInput } from "./helpers/fileReader.js";

const input = readInput("./input/day9.txt");
let num = 0;
let found = false;
for (let i = 25; i < input.length; i++) {
  num = parseInt(input[i]);
  found = false;
  for (let j = i - 25; j < i; j++) {
    for (let k = j + 1; k < i; k++) {
      if (parseInt(input[j]) + parseInt(input[k]) === num) {
        found = true;
      }
    }
  }
  if (!found) {
    console.log("Part 1:", num);
    break;
  }
}

//create array to store input values, add up input values until they are > num
//clear array, do so again starting from next input location.
//if added up values === num, find smallest and largest and add them together
const numArray = [];
found = false;
for (let i = 0; i < input.length; i++) {
  if (found) {
    break;
  }
  let total = parseInt(input[i]);
  numArray.push(total);
  for (let j = i + 1; j < input.length; j++) {
    let adder = parseInt(input[j]);
    if (total + adder < num) {
      numArray.push(adder);
      total += adder;
    } else if (total + adder > num) {
      numArray.length = 0;
      break;
    } else {
      numArray.push(adder);
      found = true;
      break;
    }
  }
}

numArray.sort();
console.log("Part 2 :", numArray[0] + numArray[numArray.length - 1]);

File: day9.py
print('Python')
file = open('./input/day9.txt')
file = [x.strip() for x in file.readlines()]


def part1():
    for i in range(25, len(file)):
        found = False
        num = int(file[i])
        for j in range(i - 25, i):
            for k in range(j + 1, i):
                if int(file[j]) + int(file[k]) == num:
                    found = True
        if not found:
            print('Part 1 : {}'.format(num))
            return num


def part2(num):
    numArray = []
    found = False
    for i in range(len(file)):
        if found:
            break
        total = int(file[i])
        numArray.append(total)
        for j in range(i + 1, len(file)):
            adder = int(file[j])
            if total + adder < num:
                numArray.append(adder)
                total += adder
            elif total + adder > num:
                numArray.clear()
                break
            else:
                numArray.append(adder)
                found = True
                break
    numArray.sort()
    print('Part 2 : {}'.format(numArray[0] + numArray[-1]))


num = part1()
part2(num)
